### PR TITLE
add npm-packager for releasing

### PR DESCRIPTION
**Pull request by yordis**
Related to #514

@rt2zz @Andarist I need your feedbacks on this.

Pretty much I enhance the npm publish and pack commands so you can specify the root of the project.

So,

What I did here was to export directly into `dist` the `commonjs` build so people could do the `require` without having to add `dist` or anything on the path.

To be honest,

I am not quite sure which one is the build that have to be flatted out (that is why I need you to double check this) but the one that we need to flat out, we can just put the output directly into `dist` folder and done.

### CODE REVIEW

**Comment by yordis** [comment]
Yeah this one, you will need to give up to one of them 🤔🤔🤔

**Comment by Andarist** [comment]
Not sure what do you mean, could you rephrase?

**Comment by yordis** [comment]
@Andarist the `commonjs` one do not need to have the `dist` on the path because it is flatted out.

Sorry for the inconvenience, I thought this may help.
